public_html/wp-content/plugins/pattern-directory/includes/pattern-post-type.php

```diff
@@ -700,6 +747,16 @@ function filter_patterns_rest_query( $args, $request ) {
 		);
 	}
 
+	$allowed_blocks = $request->get_param( 'allowed_blocks' );
+	if ( $allowed_blocks ) {
+		// Only return a pattern if all contained blocks are in the allowed blocks list.
+		$args['meta_query']['allowed_blocks'] = array(
+			'key'     => 'wpop_contains_block_types',
+			'compare' => 'REGEXP',
+			'value'   => '^((' . implode( '|', $allowed_blocks ) . '),?)+$',
```

Comment:
This isn't going to be amazing performance wise, but I'm not sure I can see a better way to achieve this, other than filtering post-query.